src/components/BigNumber/BigNumber.tsx

```diff
@@ -0,0 +1,273 @@
+							{formattedValue}
+						</View>
+						<Flex gap="size-50">
+							{cloneElement(icon, { size: iconSize })}
+							{label}
+						</Flex>
+					</View>
+				) : (
+					<>
+						{icon && (
+							<View gridArea="icon" alignSelf={iconAlign} justifySelf={iconJustify}>
+								{cloneElement(icon, { size: iconSize })}
+							</View>
+						)}
+						<View gridArea="label" alignSelf={labelAlign} UNSAFE_className="big-number-label">
+							{label}
+						</View>
+					</>
+				)}
+				{!displayCombo && (
+					<View gridArea="data" UNSAFE_className="big-number-data">
+						{formattedValue}
+					</View>
+				)}
+			</Grid>
+		</div>
+	);
+};
+
+BigNumber.displayName = 'BigNumber';
+
+function getGridProperties(
+	icon: IconElement | undefined,
+	orientation: Orientation,
+	height: number | undefined,
+	chartWidth: number,
+	lineProps?: LineProps
+) {
+	let labelAlign: 'start' | 'center' = 'start';
+	let iconAlign: 'start';
+	let iconJustify: 'end';
+	const iconSize = getIconSizeByOrientation(
+		orientation,
+		chartWidth,
+		icon !== undefined,
+		lineProps !== undefined,
+		height
+	);
+
+	// First block (line elements)
+	if (lineProps) {
+		// Then get the icon conditions
+		if (icon) {
+			if (orientation === 'vertical') {
+				labelAlign = 'center';
+				iconJustify = 'end';
+				return {
+					areas: ['sparkline', 'combo', 'data'],
+					columns: ['4fr'],
+					displayCombo: true,
+					labelAlign,
+					iconJustify,
+					iconSize,
+				};
+			}
+			if (orientation === 'horizontal') {
+				iconJustify = 'end';
+				iconAlign = 'start';
+				labelAlign = 'center';
+				return {
+					areas: ['sparkline combo'],
+					columns: ['1fr'],
+					iconJustify,
+					iconAlign,
+					displayCombo: true,
+					iconSize,
+					labelAlign,
+				};
+			}
+		} else {
+			if (orientation === 'vertical') {
+				return { areas: ['sparkline', 'data', 'label'], columns: ['1fr'], labelAlign };
+			}
+			if (orientation === 'horizontal') {
+				return { areas: ['sparkline data', 'sparkline label'], columns: ['1fr 1fr'], labelAlign };
+			}
+		}
+	}
+	// Second block (no line elements)
+	if (icon) {
+		if (orientation === 'vertical') {
+			return {
+				areas: ['icon', 'data', 'label'],
+				columns: ['1fr'],
+				iconSize,
+				labelAlign,
+			};
+		}
+		if (orientation === 'horizontal') {
+			iconJustify = 'end';
+			return {
+				areas: ['icon data', 'icon label'],
+				columns: ['1fr', '2fr'],
+				iconSize,
+				iconJustify,
+				labelAlign,
+			};
+		}
+	}
+	// Third block (default)
+	return { areas: ['data', 'label'], columns: ['1fr'], labelAlign };
+}
+
+function getIconSizeByOrientation(
+	orientation: Orientation,
+	chartWidth: number,
+	hasIcon: boolean,
+	hasLines: boolean,
+	height?: number
+): 'XXS' | 'XS' | 'S' | 'M' | 'L' | 'XL' | 'XXL' {
+	if (hasIcon) {
+		if (hasLines) {
+			if (orientation == 'vertical') {
+				return determineIconSize(height ? height / 3 : chartWidth / 2);
```

Comment:
pulling this ternary up into a variable would reduce the Sonar Cloud cognitive complexity.
```suggestion
		if (hasLines) {
		        const availableWidth = height ? height / 3 : chartWidth / 2
			if (orientation == 'vertical') {
				return determineIconSize(availableWidth);
```